### PR TITLE
Remove more tests from nose tests exclusion list

### PR DESCRIPTION
**Pull request by aaltay**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by aaltay** [comment]
R: @silviulica Please take a look.

**Comment by aaltay** [comment]
@robertwb do you know how this test should be updated?

`snippets_test.TypeHintsTest.test_deterministic_key` is failing with

```
File "/Users/travis/build/apache/incubator-beam/sdks/python/apache_beam/transforms/ptransform.py", line 584, in process_argspec_fn
raise NotImplementedError
```

Failing at applying this:

`beam.CombinePerKey(sum).with_input_types(beam.typehints.Tuple[Player, int]))`

It works if when I locally roll back #633

I think `CombinePerKey` needs to implement `process_argspec_fn` after this change.

**Comment by silviulica** [comment]
Perhaps something like https://github.com/apache/incubator-beam/blob/python-sdk/sdks/python/apache_beam/transforms/core.py#L565

**Comment by aaltay** [comment]
Thanks @silviulica. I added something similar and tests are passing. It seems like, that function is used to type check the wrapped callable.

**Comment by silviulica** [comment]
LGTM

**Comment by aaltay** [comment]
@dhalperi Please review.

**Comment by dhalperi** [comment]
LGTM

**Comment by aaltay** [comment]
Thank you.

File: sdks/python/apache_beam/transforms/core.py [hunk 1]
```diff
@@ -882,6 +882,9 @@ def make_fn(self, fn):
   def default_label(self):
     return '%s(%s)' % (self.__class__.__name__, self._fn_label)
 
+  def process_argspec_fn(self):
+    return self.fn._fn  # pylint: disable=protected-access
```
Comment (robertwb) [inline sdks/python/apache_beam/transforms/core.py]: This should probably be self.fn.pocess_argspec_fn (or something similar) as this only handles the case of CallableWrapperCombineFn.

**Comment by robertwb** [comment]
Just one comment, but other than that looks good, and nice to have all these tests again!

File: sdks/python/apache_beam/transforms/core.py (hunk 1 above)
Comment (aaltay) [inline sdks/python/apache_beam/transforms/core.py]: Thank you. This change already went in, I will fix this in a follow up.

 `CombineFn` does not expose process function similar to `DoFn`. What should it do in the default case, (base implementation for `pocess_argspec_fn`)?

**Pull request closed without merging**